File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.py
class Solution:
    def specialArray(self, nums: List[int]) -> int:
        nums.sort()
        N = len(nums)
        for i,n in enumerate(nums):
            if (N-i) <= n:
                if i==0 or (nums[i-1] != n and nums[i-1] < N-i):
                    return N-i
                return -1
        return -1

File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.java
class Solution {
    public int specialArray(int[] nums) {
        Arrays.sort(nums);
        int N = nums.length;
        int j=N;
        for(int i=0;i<N;i++){
            if(nums[i]>=N-i){
                if(i==0) return j;
                if(nums[i-1]<j) return j;
                return -1;
            }
            j--;
        }
        return -1;
    }
}